dbms/src/Common/ThreadMetricUtil.cpp

```diff
@@ -18,16 +18,18 @@
 #include <common/types.h>
 
 std::atomic<UInt64> last_max_thds_metric_reset_ts{0};
-const UInt64 max_thds_metric_reset_interval = 60; //60s
+const std::chrono::duration<UInt64> max_thds_metric_reset_interval{60}; // 60s
 
 namespace DB
 {
 bool tryToResetMaxThreadsMetrics()
 {
-    UInt64 now_ts = clock_gettime_ns_adjusted(last_max_thds_metric_reset_ts, CLOCK_MONOTONIC);
-    if (now_ts > last_max_thds_metric_reset_ts + max_thds_metric_reset_interval)
+    using microseconds = std::chrono::duration<UInt64, std::micro>;
+    UInt64 last_max_thds_metric_reset_ts_tmp = last_max_thds_metric_reset_ts.load(std::memory_order_relaxed);
+    UInt64 now_ts = clock_gettime_ns_adjusted(last_max_thds_metric_reset_ts_tmp, CLOCK_MONOTONIC);
```

Comment:
the unit of returned val is `ns`, however , we use `microseconds`

Reply:
sorry, my bad

Reply:
fixed